cardano-db-sync/src/Cardano/DbSync.hs

```diff
@@ -236,52 +234,68 @@ runDbSyncNodeNodeClient iomgr trce plugin nodeConfig (SocketPath socketPath) = d
-  :: forall blk.
-     (blk ~ ByronBlock)
-  -- TODO: the need of a 'Proxy' is an evidence that blk type is not really
-  -- needed here.  The wallet client should use some concrete type of block
-  -- from 'cardano-chain'.  This should remove the dependency of this module
-  -- from 'ouroboros-consensus'.
-  => Trace IO Text
+    dbSyncVersions txv = foldMapVersions
+        (\v ->
+          versionedNodeToClientProtocols
+            (nodeToClientProtocolVersion (Proxy @blk) v)
+            (NodeToClientVersionData { networkMagic = nodeNetworkMagic (Proxy @blk) topLevelConfig })
+            (dbSyncProtocols trce plugin topLevelConfig txv)
+        )
+        (supportedNodeToClientVersions (Proxy @blk))
+
+dbSyncProtocols
+  :: Trace IO Text
   -> DbSyncNodePlugin
   -> TopLevelConfig ByronBlock
-  -> StrictTMVar IO (GenTx blk)
-  -> ConnectionId LocalAddress
-  -> OuroborosApplication 'InitiatorApp BSL.ByteString IO () Void
-localInitiatorNetworkApplication trce plugin pInfoConfig txv _ =
-    nodeToClientProtocols NodeToClientProtocols
-      { localChainSyncProtocol =
-          -- local chain sync protocol
-          (InitiatorProtocolOnly $ MuxPeerRaw $ \channel ->
-            liftIO . logException trce "ChainSyncWithBlocksPtcl: " $ do
-              logInfo trce "Starting chainSyncClient"
-              latestPoints <- getLatestPoints
-              currentTip <- getCurrentTipBlockNo
-              logDbState trce
-              actionQueue <- newDbActionQueue
-              (metrics, server) <- registerMetricsServer
-              race_
-                (runDbThread trce plugin metrics actionQueue)
-                (runPipelinedPeer
-                      localChainSyncTracer (localChainSyncCodec @blk pInfoConfig) channel
-                      (chainSyncClientPeerPipelined (chainSyncClient trce metrics latestPoints currentTip actionQueue))
-                    )
-              atomically $
-                writeDbActionQueue actionQueue DbFinish
-              cancel server)
-        , localTxSubmissionProtocol =
-            -- local tx submission
-            (InitiatorProtocolOnly $ MuxPeer
-              (contramap (Text.pack . show) . toLogObject $ appendName "db-sync-local-tx" trce)
-              localTxSubmissionCodec
-              (localTxSubmissionClientPeer (txSubmissionClient txv)))
+  -> StrictTMVar IO (GenTx ByronBlock)
+  -> NodeToClientProtocols 'InitiatorApp BSL.ByteString IO () Void
+dbSyncProtocols trce plugin topLevelConfig txv =
+    NodeToClientProtocols {
+          localChainSyncProtocol = localChainSyncProtocol
+        , localTxSubmissionProtocol = localTxSubmissionProtocol
+        , localStateQueryProtocol = dummyLocalQueryProtocol
         }
   where
     localChainSyncTracer :: Tracer IO (TraceSendRecv (ChainSync ByronBlock (Tip ByronBlock)))
     localChainSyncTracer = toLogObject $ appendName "ChainSync" trce
 
+    codecs = clientCodecs (configBlock topLevelConfig) ByronNodeToClientVersion2
```

Comment:
We should not use `ByronNodeToClientVersion2` codecs for `ByronNodeToClientVersion1`; even though the second argument is not used currently.  If we do it properly, it will be future proof.  You can add `Codecs` to argument of `dbSyncProtocols` and use the version to which you have access in `dbSyncVersions`.

Reply:
Done